numpy/core/tests/test_arrayprint.py

```diff
@@ -524,6 +524,14 @@ def test_refcount(self):
         gc.enable()
         assert_(r1 == r2)
 
+    def test_with_sign(self):
+        a = np.array([-2, 0, 3])
+        assert_equal(
+            np.array2string(a, sign='+'),
+            '[+-2 +0 +3]'
```

Comment:
The semantics are that `+` is only added if the value is positive.

```suggestion
            '[-2 +0 +3]'
```

The results with `sign='-'` and `sign=' '` also need to be tested.

Reply:
@rkern Apologies for the oversight, thank you for the feedback.  I just submitted a fix that implements the desired behavior.

Reply:
Adding tests for sign='-' and sign=' '